File: Adding Spaces to a String.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
       char[] cStr = new char[s.length() + spaces.length];
        int i = 0; 
        int j = 0; 
        int k = 0; 
        while(i<spaces.length && j<s.length()){
            int insertionIndex = spaces[i];
            while(j<insertionIndex && j<s.length()){
                cStr[k] = s.charAt(j);
                k++;
                j++;
            }
            
            cStr[k] = ' ';
            k++;
            i++;
        }
        while(j<s.length()){
            cStr[k] = s.charAt(j);
            k++;
            j++;
        }
        return new String(cStr);
    }
}

File: week1/adding-spaces-to-a-string.py
class Solution(object):
    def addSpaces(self, s, spaces):
        """
        :type s: str
        :type spaces: List[int]
        :rtype: str
        """
        ans=[]
        last=0
        for i in spaces:
            ans.append(s[last:i])
            ans.append(" ")
            last=i
        ans.append(s[last:])
        return "".join(ans)
        